test/array.js

```diff
@@ -70,53 +70,3 @@ test('auto bool accumulates with auto string', function (t) {
 
 	t.end();
 });
-
-test('declared boolean overwrites string', function (t) {
-	var options = {
```

Comment:
why remove these tests?

Reply:
The (very edge) scenario the tests were checking is no longer reproducible.

I had previously written a weird and wonderful test to cover an edge case where the author declared a boolean option and the command-line arguments stored a string [sic] and then also stored a boolean. The test was checking the end result was storing a boolean rather than triggering the implicit array handling.

Previously this was technically possible through misuse of a short option like `-b=xyz` storing `b: 'xyz'`. This PR makes the short and long option processing consistent, so `-b=xyz` now stores `b:true` if `b` is boolean.

Reply:
Isnt that a breaking change?

Reply:
I'm currently treating it as a bonus bug fix. The behaviour was not being explicitly tested. (I have added some new tests in this PR to nail it down for the future.)

I did change the behaviour somewhat accidentally while unifying the short and long boolean behaviours to add strict mode in one place instead of many. I'm happy with how it is turning out so far, but still a work in progress.

Reply:
The change isn't essential to the implementation, and could preserve legacy inconsistent behaviour for short flags.